### PR TITLE
Deployment

### PR DESCRIPTION
**Pull request by Arslanshanii**
Added yml work flow  file for deployment

### changes made
- initialised submodule so that the files are available to access after cloning 
- added new path for cls file
- added  logic for automated building of exam file

### exam pdf automation explain

- automates the process of building exams in multiple semester subdirectories under the exam folder. It loops through each subdirectory (e.g., semester1, semester2), checks if a build.py script exists, and if found, it copies a file named examClass from the course_template folder into the current subdirectory. Then, it navigates into the subdirectory, runs the build.py script using Python, and returns to the original directory. If no build.py is found, it skips that directory.

### CODE REVIEW

File: .github/workflows/BuildPDFs.yml [hunk 1]
```diff
@@ -13,16 +13,35 @@ jobs:
...
+            for dir in exam/*/; do
+            if [ -f "$dir/build.py" ]; then
+              echo "Building in $dir"
+              cp ../../course_template/examClass "$dir"
+              cd "$dir" && python3 build.py && cd - 
+            else
+              echo "Skipping $dir: no build.py found."
+            fi
+            done
+
```
Comment (wallscheid) [inline .github/workflows/BuildPDFs.yml]: see other PR comment: this workflow will only work if we add (manually configured) build files per exam subfolder - I would appreciate a more automated routine

**Review by wallscheid — CHANGES_REQUESTED** [review]
Why does this PR also include the exam build files as they are already addressed in the other parallel PR? Anyhow, the main issue regarding the exam automation remains present

File: .github/workflows/BuildPDFs.yml (hunk 1 above)
Comment (Arslanshanii) [reply .github/workflows/BuildPDFs.yml]: Added a new build.py which will loop through the subdirectories and will automate the process of building pdfs,
Yaml file is changed accordingly .

**Review by wallscheid — APPROVED** [review]
Looks good - let's give it a try with the new deployment workflow

**Pull request merged**